File: frontend/src/utils/mediaRegions.ts
export type MediaRegion =
  | 'western'
  | 'mainland'
  | 'hong-kong-taiwan'
  | 'korea'
  | 'japan'
  | 'asia-pacific'

export interface MediaRegionOption {
  value: MediaRegion
  label: string
  countryCodes: readonly string[]
}

export const NEXTFIND_MEDIA_REGIONS: readonly MediaRegionOption[] = [
  {
    value: 'western',
    label: '欧美',
    countryCodes: ['US', 'GB', 'FR', 'DE', 'IT', 'ES', 'CA', 'AU', 'RU', 'BR', 'MX', 'AR'],
  },
  { value: 'mainland', label: '大陆', countryCodes: ['CN'] },
  { value: 'hong-kong-taiwan', label: '港台', countryCodes: ['HK', 'TW', 'MO'] },
  { value: 'korea', label: '韩国', countryCodes: ['KR'] },
  { value: 'japan', label: '日本', countryCodes: ['JP'] },
  {
    value: 'asia-pacific',
    label: '亚太',
    countryCodes: ['IN', 'TH', 'SG', 'MY', 'PH', 'VN', 'ID', 'MM'],
  },
]

export function formatMediaRegions(countryCodes: string[] | null | undefined): string {
  if (!countryCodes?.length) return ''
  const normalized = new Set(countryCodes.map((code) => code.trim().toUpperCase()))
  const labels = NEXTFIND_MEDIA_REGIONS
    .filter((region) => region.countryCodes.some((code) => normalized.has(code)))
    .map((region) => region.label)
  return labels.length ? labels.join(' / ') : '其他'
}

File: backend/app/core/media_regions.py
from __future__ import annotations

from enum import StrEnum


class MediaRegion(StrEnum):
    WESTERN = "western"
    MAINLAND = "mainland"
    HONG_KONG_TAIWAN = "hong-kong-taiwan"
    KOREA = "korea"
    JAPAN = "japan"
    ASIA_PACIFIC = "asia-pacific"


NEXTFIND_REGION_COUNTRY_CODES: dict[MediaRegion, tuple[str, ...]] = {
    MediaRegion.WESTERN: (
        "US",
        "GB",
        "FR",
        "DE",
        "IT",
        "ES",
        "CA",
        "AU",
        "RU",
        "BR",
        "MX",
        "AR",
    ),
    MediaRegion.MAINLAND: ("CN",),
    MediaRegion.HONG_KONG_TAIWAN: ("HK", "TW", "MO"),
    MediaRegion.KOREA: ("KR",),
    MediaRegion.JAPAN: ("JP",),
    MediaRegion.ASIA_PACIFIC: (
        "IN",
        "TH",
        "SG",
        "MY",
        "PH",
        "VN",
        "ID",
        "MM",
    ),
}


def country_codes_for_region(region: MediaRegion) -> tuple[str, ...]:
    return NEXTFIND_REGION_COUNTRY_CODES[region]
